### PR TITLE
Support for ot-rcp-spi

### PR DESCRIPTION
**Pull request by parag-silabs**
- Add a new project file (openthread-efr32-rcp-spi.slcp) to generate openthread-efr32-rcp-spi and openthread-efr32-rcp-spi-mbedtls libs.
- Add a patch file to apply over gecko_sdk for ot-rcp-spi target to generate libs and reset head back after generation.
- Update build script to build 'ot-rcp-spi' image.
- Re-organize build directories to compile UART and SPI ot-rcp targets into respective dirs.
- Clean up project files.

### CODE REVIEW

**Comment by parag-silabs** [comment]
@romacdon @lmnotran @suveshpratapa @bertoldi-silabs Can you please review the PR?

File: CMakeLists.txt [hunk 1]
```diff
@@ -81,7 +81,7 @@ set(OT_MBEDTLS ${OT_EXTERNAL_MBEDTLS})
 
 # RCP mbedtls library
 if(NOT OT_MBEDTLS_RCP)
-    set(OT_MBEDTLS_RCP "openthread-efr32-rcp-mbedtls" CACHE STRING "default efr32 RCP mbedtls library" FORCE)
+    set(OT_MBEDTLS_RCP "openthread-efr32-rcp-uart-mbedtls" CACHE STRING "default efr32 RCP mbedtls library" FORCE)
```
Comment (romacdon) [inline CMakeLists.txt:84]: Do we need a `openthread-efr32-rcp-spi-mbedtls` library or should this remain as `openthread-efr32-rcp-mbedtls` and its used for both spi and uart?

File: CMakeLists.txt [hunk 2]
```diff
@@ -64,7 +64,7 @@ endif()
 
 # RCP platform library
 if(NOT OT_PLATFORM_LIB_RCP)
-    set(OT_PLATFORM_LIB_RCP "openthread-efr32-rcp" CACHE STRING "default efr32 RCP platform library" FORCE)
+    set(OT_PLATFORM_LIB_RCP "openthread-efr32-rcp-uart" CACHE STRING "default efr32 RCP platform library" FORCE)
```
Comment (romacdon) [inline CMakeLists.txt:67]: Do we need a `openthread-efr32-rcp-spi` library or should this remain as `openthread-efr32-rcp` and its used for both spi and uart?

Comment (romacdon) [reply CMakeLists.txt:67]: Actually I see this comment in `script/build` 
``` 
# ==============================================================================================================
# Generate openthread-efr32-rcp-spi and openthread-efr32-rcp-spi-mbedtls libs
# ==============================================================================================================
```
So don't we need `openthread-efr32-rcp-spi`.

File: CMakeLists.txt (hunk 1 above)
Comment (romacdon) [reply CMakeLists.txt:84]: See comment above.  Don't we need `openthread-efr32-rcp-spi-mbedtls`?

**Review by romacdon — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (parag-silabs) [reply CMakeLists.txt:84]: That is default config set to uart one. We do set correct libs in build script.
We can do something like

if(NOT OT_MBEDTLS_RCP)
      if(OT_NCP_SPI) 
          set spi
      else 
         set uart one.
     endif
endif

File: CMakeLists.txt (hunk 2 above)
Comment (parag-silabs) [reply CMakeLists.txt:67]: Same comment applies here too.

File: CMakeLists.txt (hunk 1 above)
Comment (romacdon) [reply CMakeLists.txt:84]: I see.  Although it may not be needed I think it would more clear to add the logic you suggested.

Comment (parag-silabs) [reply CMakeLists.txt]: Sounds good 👍, I just pushed the changes.

**Review by romacdon — APPROVED** [review]
LGTM

**Review by lmnotran — APPROVED** [review]

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by bertoldi-silabs — APPROVED** [review]

**Comment by parag-silabs** [comment]
@jwhui can you please have review the PR?

**Review by jwhui — APPROVED** [review]

**Review by hemanth-silabs — COMMENTED** [review]
LGTM 👍

**Pull request merged**